extension/src/nmost/max_by_float.rs

```diff
@@ -0,0 +1,220 @@
+use pgx::{iter::TableIterator, *};
+
+use crate::nmost::max_float::toolkit_experimental::*;
+use crate::nmost::*;
+
+use crate::{
+    build, flatten,
+    palloc::{Internal, InternalAsValue, ToInternal},
+    pg_type, ron_inout_funcs,
+};
+
+use ordered_float::NotNan;
+use std::cmp::Reverse;
+
+type MaxByFloatTransType = NMostByTransState<Reverse<NotNan<f64>>>;
+
+#[pg_schema]
+pub mod toolkit_experimental {
+    use super::*;
+
+    pg_type! {
+        #[derive(Debug)]
+        struct MaxByFloats<'input> {
+            values: MaxFloatsData<'input>,  // Nesting pg_types adds 8 bytes of header
+            data: DatumStore<'input>,
+        }
+    }
+    ron_inout_funcs!(MaxByFloats);
+
+    impl<'input> From<MaxByFloatTransType> for MaxByFloats<'input> {
+        fn from(item: MaxByFloatTransType) -> Self {
+            let (capacity, val_ary, data) = item.into_sorted_parts();
+            unsafe {
+                flatten!(MaxByFloats {
+                    values: build!(MaxFloats {
+                        capacity: capacity as u32,
+                        elements: val_ary.len() as u32,
+                        values: val_ary
+                            .into_iter()
+                            .map(|x| f64::from(x.0))
+                            .collect::<Vec<f64>>()
+                            .into()
+                    })
+                    .0,
+                    data,
+                })
+            }
+        }
+    }
+}
+
+#[pg_extern(schema = "toolkit_experimental", immutable, parallel_safe)]
```

Comment:
Stylistic question:  why do we sometimes do this rather than just move all these functions into the toolkit_experimental module?  It's always confused me that we use the two different ways to put something into a schema.

Reply:
You know, I've actually just sort of always followed the existing code and not really thought about it.  I had thought that the schema name was needed for functions to make pgx put them in the right space, but it seems to do the right thing if we just put it in the module as well.  Maybe this wasn't as reliable in a previous version of pgx?